packages/server/migrations/20240918183915_add_name_to_arpa_subrecipients.js

```diff
@@ -0,0 +1,24 @@
+/**
+ * @param { import("knex").Knex } knex
+ * @returns { Promise<void> }
+ */
+exports.up = async function (knex) {
+    await knex.schema.table('arpa_subrecipients', (table) => {
+        table.text('name');
+    });
+    await knex.raw(
+        'CREATE UNIQUE INDEX idx_arpa_subrecipients_tenant_id_name_unique ON arpa_subrecipients (tenant_id, name) WHERE name is not null and uei is null and tin is null',
```

Comment:
> i.e. should duplicate non-null names for the same tenant allowed in any circumstance, regardless of uei or tin?

Right now we're working with insufficient data here. Working assumption is that there _could_ be subrecipients in the system that share the same name but have slightly different UEI/TIN since this was never enforced in the past.

> Also we may want an additional constraint to ensure that at least one of uei/tin/name is provided. If that sounds like a good idea to you, I think that could be implemented as a CHECK constraint like:

This makes sense to add since we enforce this on the application layer, can translate to a database check as well.

Reply:
Resolved